datalad/support/network.py

```diff
@@ -969,38 +1007,43 @@ def is_ssh(ri):
         or (isinstance(_ri, URL) and _ri.scheme == 'ssh')
 
 
-def get_local_file_url(fname, compatibility='git-annex'):
+def get_local_file_url(fname: str,
+                       compatibility: str = 'git-annex',
+                       allow_relative_path: bool = True
+                       ) -> str:
     """Return OS specific URL pointing to a local file
 
     Parameters
     ----------
     fname : string
         Filename.  If not absolute, abspath is used
-    compatibility : {'git', 'git-annex'}, optional
-        On Windows, and only on that platform, file:// URLs may need to look
-        different depending on the use case or consuming application. This
-        switch selects different compatibility modes: 'git' for use with
-        Git commands (e.g. `clone` or `submodule add`); 'git-annex` for
-        Git-annex command input (e.g. `addurl`). On any other platform this
-        setting has no effect.
+    compatibility : str, optional
+        This parameter is only interpreted on Windows systems. If set to
+        anything else than 'git', the anchor, e.g. `C:` of `fname` will be put
+        into the `file-auth` part, i.e. network location, defined in RFC 8089.
+        This option is mainly used to support git-annex specific encoding of
+        Windows paths.
+    allow_relative_path: bool, optional
+        Allow `fname` to be a relative path. The path will be converted to an
+        absolute path, by using the current directory as path prefix.
     """
-    # we resolve the path to circumwent potential short paths on unfortunate
-    # platforms that would ruin the URL format.
-    #path = Path(fname).resolve().absolute()
-    path = Path(fname).resolve()
-    if on_windows:
-        # in addition we need to absolute(), as on windows resolve() doesn't
-        # imply that
-        path = path.absolute().as_posix()
-        furl = 'file://{}{}'.format(
-            '/' if compatibility == 'git' else '',
-            urlquote(
-                re.sub(r'([a-zA-Z]):', r'\1', path)
-            ))
-    else:
-        # TODO:  need to fix for all the encoding etc
-        furl = str(URL(scheme='file', path=str(path)))
-    return furl
+    path_obj = Path(fname)
+    if not path_obj.is_absolute():
+        if allow_relative_path is True:
```

Comment:
```suggestion
        if allow_relative_path:
```
to not suggest some other special `True` value (like non empty list etc) handling.